ui-components/src/page_components/listing/ListingCard.tsx

```diff
@@ -22,10 +26,80 @@ export interface ListingCardProps {
   tableHeaderProps?: ListingCardHeaderProps
   tableProps: ListingCardTableProps
   detailsLinkClass?: string
+  listingID: string
 }
 
 const ListingCard = (props: ListingCardProps) => {
   const { imageCardProps, tableProps, detailsLinkClass, tableHeaderProps } = props
+  const { profile, userProfileService } = useContext(AuthContext)
+  const preferences: UserPreferences = profile?.preferences || {
+    sendEmailNotifications: false,
+    sendSmsNotifications: false,
+    favoriteIDs: [],
+  }
+  const [favoriteState, updateFavoriteState] = useState("favored")
+
+  useEffect(() => {
+    if (preferences.favoriteIDs?.includes(props.listingID)) {
+      updateFavoriteState("favored")
+    } else {
+      updateFavoriteState("notFavored")
+    }
+  }, [preferences.favoriteIDs, props.listingID])
+
+  const addToFavorite = async () => {
+    if (!profile) {
+      return
+    }
+    updateFavoriteState("favored")
```

Comment:
You should move this to after the `userProfileService?.update()` below so if writing to the backend fails we won't update the UI